integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/exception/WebClientResponseWithMessageException.java

```diff
@@ -8,21 +8,37 @@
  *
  * <p>In this class we override the {@link #getMessage()} behavior, to append server response if it
  * exists. It doesn't change status code (ex: forbidden) or status int (ex. 403)
+ *
+ * <p>An exception thrown in Openhouse clients to indicate an error in a tables API request, acting
+ * as a wrapper around WebClientResponseException.
  */
-public class WebClientResponseWithMessageException extends WebClientResponseException {
-  public WebClientResponseWithMessageException(WebClientResponseException e) {
-    super(
+public class WebClientResponseWithMessageException extends WebClientWithMessageException {
```

Comment:
Don't we need to expose some of the critical information, at _minimum_ `getStatusCode()` (as an `int`, to avoid exposing `HttpStatusCode`)? This is crucial for callers to understand how to handle the exception (e.g. is it retryable, is it a user issue vs network issue vs ...)

Reply:
yeah, that makes sense. I can add `getStatusCode()` to expose the status code and possibly `getStatusText()` to expose status text in a separate PR